### ISSUE TITLE
Node have put object in one shard then try to remove from another shard

### ISSUE THREAD

**Issue by anikeev-yadro** [nspcc-dev/neofs-node#2266]
I saw that object was PUT to shard `8HPVZ4ETzwTP9iBT9y959A`
```
2023-02-21T02:58:34.099Z	info	log/log.go:13	local object storage operation	{"shard_id": "8HPVZ4ETzwTP9iBT9y959A", "address": "B13hwjP8iEEhWbHrAnRmasb9KDcPXhvxNbA2umBesSMz/EMUB5U2traWn8nA2c8u3Yuwuwkm2SLqCsCsPyjKtr58x", "op": "metabase PUT"}
```
then node tried to remove this object from another shard `FBYYAY8qX4kkEMVJqUPhWU`
```
2023-02-21T03:04:14.258Z	info	log/log.go:13	local object storage operation	{"shard_id": "FBYYAY8qX4kkEMVJqUPhWU", "address": "B13hwjP8iEEhWbHrAnRmasb9KDcPXhvxNbA2umBesSMz/EMUB5U2traWn8nA2c8u3Yuwuwkm2SLqCsCsPyjKtr58x", "op": "metabase DELETE"}
2023-02-21T03:04:14.422Z	debug	shard/delete.go:92	can't remove object from blobStor	{"shard_id": "FBYYAY8qX4kkEMVJqUPhWU", "object_address": "B13hwjP8iEEhWbHrAnRmasb9KDcPXhvxNbA2umBesSMz/EMUB5U2traWn8nA2c8u3Yuwuwkm2SLqCsCsPyjKtr58x", "error": "status: code = 2049 message = object not found"}
```
And one hour later node failed to replicate this object 
```
2023-02-21T04:02:34.653Z	debug	policer/check.go:231	shortage of object copies detected	{"component": "Object Policer", "object": "B13hwjP8iEEhWbHrAnRmasb9KDcPXhvxNbA2umBesSMz/EMUB5U2traWn8nA2c8u3Yuwuwkm2SLqCsCsPyjKtr58x", "shortage": 1}
2023-02-21T04:02:34.662Z	error	replicator/process.go:62	could not replicate object	{"component": "Object Replicator", "node": "02ac920cd7df0b61b289072e6b946e2da4e1a31b9ab1c621bb475e30fa4ab102c3", "object": "B13hwjP8iEEhWbHrAnRmasb9KDcPXhvxNbA2umBesSMz/EMUB5U2traWn8nA2c8u3Yuwuwkm2SLqCsCsPyjKtr58x", "error": "(*putsvc.RemoteSender) could not send object: (*putsvc.remoteTarget) could not put object to [/dns4/s03.neofs.devenv/tcp/8080]: write object via client: status: code = 1024 message = object has expired"}
```
## Logs
[obj-nn-test-node2.spb.yadro.com-s01-log.txt.zip](https://github.com/nspcc-dev/neofs-node/files/10794809/obj-nn-test-node2.spb.yadro.com-s01-log.txt.zip)
[obj-nn-test-node2.spb.yadro.com-s02-log.txt.zip](https://github.com/nspcc-dev/neofs-node/files/10794812/obj-nn-test-node2.spb.yadro.com-s02-log.txt.zip)
[obj-nn-test-node2.spb.yadro.com-s03-log.txt.zip](https://github.com/nspcc-dev/neofs-node/files/10794815/obj-nn-test-node2.spb.yadro.com-s03-log.txt.zip)
[obj-nn-test-node2.spb.yadro.com-s04-log.txt.zip](https://github.com/nspcc-dev/neofs-node/files/10794816/obj-nn-test-node2.spb.yadro.com-s04-log.txt.zip)
[obj-nn-test-node2.spb.yadro.com-s3_gate-log.txt.zip](https://github.com/nspcc-dev/neofs-node/files/10794817/obj-nn-test-node2.spb.yadro.com-s3_gate-log.txt.zip)

## Version
```
0.35.0-47-gdea97133
```
## Your Environment

DEVENV

**Comment by roman-khimov** [comment]
I think this one was fixed in #2821, @carpawell?

**Comment by carpawell** [comment]
Yes, #2821 made it obsolete.
